### PR TITLE
Added dropdown buttons to Output Window to show/hide R commands/comments

### LINKED ISSUE #8444: Add to toolbar to facilitate showing R commands
@N-thony is this a possible task for @derekagorhom?  @Patowhiz can help with the code, if this is needed.

Here is the current toolbar with the current down-arrow for the log/script window.  It is ticked when the log/script is open and not otherwise.

image

Could you add 2 further entries underneath that
```
Show R Commands
Show Comments
```
@Patowhiz as I write this I wonder whether the placing is correct in the toolbar.  Instead I wonder about adding a new down arrow after the Output icon, and putting them there?  That seems more logical?

When R commands are shown in the output window, then this is ticked.  Likewise with the comments.
When you click on either of them it doesn't necessarily change those settings. Instead, it opens the T`ools > Options > Output Window `tab, (see below) so you can change the settings.  Then the toolbar items are ticked or unticked, depending on those settings.

image

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes #8444 

I have added `Show R commands` and `include comments` to the Output tab.

@rdstern , @N-thony can you review this

Thanks

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@derekagorhom well done, I like this.

a) The settings seem to link perfectly with the Tools menu. 
b) I occasionally switched off commands and it seemed to switch off comments too?  But I couldn't get that to repeat consistently?
c) The message in the output window seems odd.  I think it is the same as when you use the tools menu.  I wonder if, instead, it could simply state what the current setting now is, or what has been done?  For example:  `Comments from dialogs displayed in the output window.  
`or `Comments from dialogs not displayed`
Similarly `R commands displayed in the output window`
or `R commands not displayed. (They continue to be copied into the log window.)`

Please @Patowhiz could you also check.  This is not quite what you requested, which was simply to open the Tools > Options dialog at the right place.  However, I find this to be simpler and nicely synchronised?

File: instat/dlgOptions.vb [hunk 1]
```diff
@@ -359,6 +359,8 @@ Public Class dlgOptions
...
         ApplyEnabled(True)
+        frmMain.mnuShowRCommand.Checked = ucrChkShowRCommandsinOutputWindow.chkCheck.Checked
+        frmMain.mnuIncludeComments.Checked = ucrChkIncludeCommentsbyDefault.chkCheck.Checked
     End Sub
```
Comment (Patowhiz) [inline instat/dlgOptions.vb:364]: Can line 362 and 363 be moved to below line 843 of `frmMain`. All we need is to change the checked status after `dlgOptions` has been closed.

File: instat/frmMain.vb [hunk 2]
```diff
@@ -642,6 +644,20 @@ Public Class frmMain
...
+        Me.clsInstatOptions.SetCommandInOutpt(mnuShowRCommand.Checked)
+        dlgOptions.ucrChkShowRCommandsinOutputWindow.chkCheck.Checked = mnuShowRCommand.Checked
+        Me.clsInstatOptions.ExecuteRGlobalOptions()
+    End Sub
+
+    Private Sub mnuIncludeComments_Click(sender As Object, e As EventArgs) Handles mnuIncludeComments.Click
+        mnuIncludeComments.Checked = Not mnuIncludeComments.Checked
+        Me.clsInstatOptions.SetIncludeCommentByDefault(mnuIncludeComments.Checked)
+        dlgOptions.ucrChkIncludeCommentsbyDefault.chkCheck.Checked = mnuIncludeComments.Checked
+        Me.clsInstatOptions.ExecuteRGlobalOptions()
```
Comment (Patowhiz) [inline instat/frmMain.vb:658]: I don't think we need to have line 657. I think the `dlgOptions` dialog always reads from `clsInstatOptions` object when it's launched.
The same applies to line 650.

**Comment by Patowhiz** [comment]
@rdstern I agree it's not what I proposed. But if you are happy with having this feature done in this way, I have no strong objection.

@derekagorhom changing of the comments can be done at the `clsInstatOptions` level. I recommend doing it at that level to maintain the single responsibility principle.

**Review by rdstern — DISMISSED** [review]
@derekagorhom I am now happy this.  Many thanks.
@lloyddewit over to you

File: instat/clsInstatOptions.vb [hunk 3]
```diff
@@ -315,6 +317,22 @@ Imports RDotNet
...
+        'add "R.commands.displayed.in.the.output.window" as options parameter of its been changed
+        strRCommands = GetROption("R.commands.displayed.in.the.output.window")
+        If frmMain.mnuShowRCommand.Checked Then
+            clsOptionsFunction.AddParameter("R.commands.displayed.in.the.output.window", "TRUE")
+        Else
+            clsOptionsFunction.AddParameter("R.commands.displayed.in.the.output.window", "FALSE")
+        End If
+
+        'add "Comments.from.dialogs.displayed.in.the.output.window" as options parameter of its been changed
+        strRComments = GetROption("Comments.from.dialogs.displayed.in.the.output.window")
```
Comment (lloyddewit) [inline instat/clsInstatOptions.vb:329]: needed?
```suggestion
```

File: instat/clsInstatOptions.vb [hunk 4]
```diff
@@ -294,6 +294,8 @@ Imports RDotNet
     Public Sub ExecuteRGlobalOptions()
         Dim clsOptionsFunction As New RFunction
         Dim strROption As String
+        Dim strRCommands As String
+        Dim strRComments As String
```
Comment (lloyddewit) [inline instat/clsInstatOptions.vb:298]: Do these variable have any purpose? delete?
```suggestion
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@derekagorhom Thanks, looks good, just a few questions

**Comment by derekagorhom** [comment]
@lloyddewit , i have made the changes can you review them
thanks

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Since your last review, the code has changed a little. In theory the functionality should be unchanged but it would be good to do a sanity check. If you can test/approve, then we can merge.
Thanks

**Review by rdstern — APPROVED** [review]
@lloyddewit and @derekagorhom looks fine.  I like it!

**Comment by Patowhiz** [comment]
Hello @derekagorhom,

I appreciate the progress you've made in this pull request. The inclusion of shortcuts is undoubtedly a significant enhancement that will greatly enhance usability.

To @lloyddewit and @rdstern, I noticed an anomaly in the output window upon launching the application. It think some of the code modifications introduced in this PR may have contributed to this issue.

I'd like to highlight that the display of comments and commands within R-Instat is not intrinsically an R based feature, as indicated by the introduction of parameters to the options R command at lines 320 and 327. Rather, this implementation operates at the front-end level (VB.Net). The options R command typically does not accommodate such parameters. This discrepancy could potentially appear odd to R users.

@derekagorhom, I understand your intention to address @rdstern's suggestion, which pertains to including a comment in the output window indicating the status of comments and commands being turned on or off. @rdstern, please feel free to provide clarification if my interpretation deviates from your original intent.

@rdstern if you agree with my assessment of your suggestion, I would recommend rectifying this matter by either amending the implementation to align with the intended behavior or considering the removal of those parameters from the R command. 

@derekagorhom if you agree with my code assessment I've highlighted the specific lines that require modification in the attached code screenshot. 

Kindly note, this discrepancy is not a critical issue warranting immediate attention. I acknowledge any delay in my review and apologize for it.

Thank you for your valuable contributions.

Output window screenshot.
delete

Code screenshot
delete2

**Comment by lloyddewit** [comment]
@Patowhiz thank you for the comment. If we leave the comment in a merged PR, then there's a risk it will be forgotten. I moved your comment to issue #8517.

**Pull request merged**